### PR TITLE
Anthropic: Chat completion & Claude 3 (Opus)

### LINKED ISSUE #4148: Add support for Claude 3

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes: https://github.com/dust-tt/dust/issues/4148

- core: Anthropic chat completion + streaming
- core: Fix mistral context size computation
- front: Add opus and sonnet latest model (Dust apps) (new chat/non chat distinction)
- front: Add claude-3 (both as global agent and available model to build assistants)
- front: Add claude-3-sonnet (global agent only)
- front: Fixes a few bugs around agents seen here and there.
 
This PR does not remove claude-2 or claude-instant. It does rename @claude into @claude-2 but that's just a pure front-end renaming (sId are preserved, renaming impacts all past and future conversations, no migration)
 
## Risk

N/A

## Deploy Plan

- deploy `core` (this will break claude 2 (not 2.1 so ~no-one) chat access)
- deploy `front`

### CODE REVIEW

File: core/src/providers/anthropic.rs [hunk 1]
```diff
@@ -71,164 +216,367 @@ impl AnthropicLLM {
...
-                match max_tokens {
-                    Some(m) => m,
-                    None => 256,
-                },
-                temperature,
-                top_p,
-                None,
-                stop_tokens.as_ref(),
-            )
+        println!("body: {}", body);
```
Comment (fontanierh) [inline core/src/providers/anthropic.rs:259]: nit cleanup

File: core/src/providers/anthropic.rs [hunk 2]
```diff
@@ -597,36 +947,123 @@ impl LLM for AnthropicLLM {
...
+                    Some(c) => Some(c.clone()),
+                    None => None,
+                },
+                _ => None,
+            },
+            None => None,
+        };
+
+        if system.is_some() {
+            println!("system: {:?}", system);
```
Comment (fontanierh) [inline core/src/providers/anthropic.rs:968]: do we want to keep this log ?

**Review by fontanierh — APPROVED** [review]
Looks great !

**Review by flvndvd — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀

File: core/src/providers/anthropic.rs (hunk 2 above)
Comment (spolu) [reply core/src/providers/anthropic.rs:968]: ah no good catch

File: core/src/providers/anthropic.rs (hunk 1 above)
Comment (spolu) [reply core/src/providers/anthropic.rs:259]: thanks

**Pull request merged**